InnerEye/ML/Histopathology/models/deepmil.py

```diff
@@ -187,8 +191,11 @@ def log_metrics(self,
                 log_on_epoch(self, f'{stage}/{metric_name}', metric_object)
 
     def forward(self, images: Tensor) -> Tuple[Tensor, Tensor]:  # type: ignore
-        with no_grad():
+        if self.is_finetune:
             H = self.encoder(images)                        # N X L x 1 x 1
+        else:
+            with no_grad():
```

Comment:
As a cleaner alternative to `if/else`, consider using `with torch.set_grad_enabled(self.is_finetune):` ([docs](https://pytorch.org/docs/stable/generated/torch.set_grad_enabled.html)).